languages/perl/duplicate_file_finder.pl

```diff
@@ -0,0 +1,680 @@
+# initial script for finding and comparing duplicate files
+
+use strict;
+use warnings;
+
+use Getopt::Std;
+
+use File::Spec;
+use File::Find;
+use File::Compare;
+use Cwd 'abs_path';
+use POSIX qw/strftime/;
+use File::Copy 'move';
+use Data::UUID;
+
+use Data::Dumper;
```

Comment:
Since this is just a simple demo script, I didn't setup proper dependency or runtime management. We could containerized this and had it runnable in Docker, but for now, I just installed packages manually:

https://www.cpan.org/modules/index.html